pkg/deploy/deploy.go

```diff
@@ -71,8 +71,8 @@ func New(ctx context.Context, log *logrus.Entry, config *RPConfig, version strin
 	return &deployer{
 		log: log,
 
-		globaldeployments:      features.NewDeploymentsClient(config.Configuration.GlobalSubscriptionID, authorizer),
-		globalrecordsets:       dns.NewRecordSetsClient(config.Configuration.GlobalSubscriptionID, authorizer),
+		globaldeployments:      features.NewDeploymentsClient(StringOrNil(config.Configuration.GlobalSubscriptionID), authorizer),
```

Comment:
These should never be nil, or empty. This makes all these calls very overloaded. Please remove